vignettes/meetupr.Rmd

```diff
@@ -0,0 +1,223 @@
+---
+title: "Setting up and Getting Started with meetupr"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Setting up and Getting Started with meetupr}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r}
+#| include: false
+#| label: setup
+
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  eval = TRUE
+)
+
+vcr::setup_knitr(prefix = "meetupr-")
+meetupr:::mock_if_no_auth()
+Sys.setenv(MEETUPR_DEBUG = "")
+```
+
+```{r}
+#| label: load
+library(meetupr)
+```
+
+## Quick Setup Check
+
+Start by running `meetup_sitrep()` to set up authentication or check your authentication status:
+
+```{r}
+#| label: sitrep
+#| cassette: true
+
+meetup_sitrep()
+```
+
+This function will show you:  
+- Which authentication method is active  
+- Whether your credentials are properly configured  
+- API connectivity status  
+- Setup instructions if authentication is missing  
+
+## Authentication Methods
+
+meetupr supports two authentication methods:
+
+### OAuth Authentication (Interactive)
+
+Best for interactive development, data exploration, and one-off analyses.
+
+**Setup:**
+```r
+# Authenticate interactively (opens browser)
+meetup_auth()
+
+# Check authentication status
+meetup_auth_status()
+```
+
+The OAuth token is cached locally in your httr2 cache directory. 
+Token files follow the pattern: `{hash}-token.rds.enc`
+
+### CI/CD Authentication (Non-Interactive)
+
+For automated workflows, you can set up token-based authentication:
+
+**1. Authenticate locally first:**
+```r
+meetup_auth()
+meetup_auth_setup_ci()
+```
+
+This encodes your token as base64 and provides instructions for setting CI secrets.
+
+**2. Set environment variables in your CI:**
+- `MEETUP_TOKEN` - Base64-encoded token (from step 1)
+- `MEETUP_TOKEN_FILE` - Token filename (e.g., `ae743e0-token.rds.enc`)
+
+**3. Token loads automatically in CI** 
+
+To make the authentication work in CI/CD pipelines, add this line before running any API calls:
+
+```r
+meetup_auth_load_ci()
+```
+
+
+
+## Environment Variables Reference
+
+| Variable | Required | Purpose |
+|----------|----------|---------|
+| `MEETUP_CLIENT_ID` | Optional* | OAuth client ID |
+| `MEETUP_CLIENT_SECRET` | Optional* | OAuth client secret |
+| `MEETUP_CLIENT_NAME` | No | OAuth client name (default: "meetupr") |
```

Comment:
what is the difference between No and Optional?

Reply:
none? removed "Optional"